src/PrestaShopBundle/DependencyInjection/Compiler/CommandAndQueryCollectorPass.php

```diff
@@ -59,9 +59,10 @@ private function findCommandsAndQueries(ContainerBuilder $container)
         $handlers = $container->findTaggedServiceIds('tactician.handler');
 
         $commands = [];
-        foreach ($handlers as $handler) {
+        foreach ($handlers as $id => $handler) {
+            $definition = $container->getDefinition($id);
             if (isset(current($handler)['command'])) {
-                $commands[] = current($handler)['command'];
+                $commands[$definition->getClass()] = current($handler)['command'];
```

Comment:
About splitting by scope I think we can find it based on the command FQCN

```suggestion
                $command = current($handler)['command'];
                preg_match('/PrestaShop\\\\PrestaShop\\\\Core\\\\Domain\\\\([a-zA-Z]+)\\\\/', $command, $matches);
                $domain = $matches[1];
                $commands[] = [
                    'handler' => $definition->getClass(),
                    'domain' => $domain,
                    'command' => $command,
                ]
```

Reply:
Can't do anything until we have a clear vision of what we want to achieve and pr scope is settled :stuck_out_tongue: (just because there are lots of improvements that can be made and it depends on dev opinion/personal taste etc.)

Reply:
Yes besides it can be done in the parser (as you did)